evm/auxiliary/user_account_contract/transaction.py

```diff
@@ -0,0 +1,210 @@
+from eth_keys import keys
+
+from evm.vm.forks.sharding.transactions import (
+    ShardingTransaction,
+)
+
+from evm.validation import (
+    validate_uint256,
+    validate_canonical_address,
+    validate_is_bytes,
+)
+
+from evm.utils.transactions import (
+    V_OFFSET,
+)
+from evm.utils.numeric import (
+    big_endian_to_int,
+    int_to_big_endian,
+)
+from evm.utils.padding import (
+    pad32,
+)
+
+from .contract import (
+    CALLDATA_SIGNATURE,
+    CALLDATA_V,
+    CALLDATA_R,
+    CALLDATA_S,
+    CALLDATA_NONCE,
+    CALLDATA_GASPRICE,
+    CALLDATA_VALUE,
+    CALLDATA_MIN_BLOCK,
+    CALLDATA_MAX_BLOCK,
+    CALLDATA_DESTINATION,
+    CALLDATA_DATA,
+)
+
+
+def slice32(start):
+    return slice(start, start + 32)
+
+
+SIGNATURE_SLICE = slice(CALLDATA_SIGNATURE, CALLDATA_SIGNATURE + 96)
+SIGNATURE_V_SLICE = slice32(CALLDATA_V)
+SIGNATURE_R_SLICE = slice32(CALLDATA_R)
+SIGNATURE_S_SLICE = slice32(CALLDATA_S)
+NONCE_SLICE = slice32(CALLDATA_NONCE)
+GASPRICE_SLICE = slice32(CALLDATA_GASPRICE)
+VALUE_SLICE = slice32(CALLDATA_VALUE)
+MIN_BLOCK_SLICE = slice32(CALLDATA_MIN_BLOCK)
+MAX_BLOCK_SLICE = slice32(CALLDATA_MAX_BLOCK)
+DESTINATION_SLICE = slice32(CALLDATA_DESTINATION)
+DATA_SLICE = slice(CALLDATA_DATA, None)  # open ended
+
+EMPTY_DATA = b'\x00' * DESTINATION_SLICE.stop
+
+
+class ForwardingTransaction(ShardingTransaction):
```

Comment:
Other name ideas

- `StandardAccountTransaction`
- `ProxyTransaction`
- `EIP86Transaction` (or whatever the appropriate EIP number is)

Reply:
`StandardAccountTransaction` was the name before I switched to `ForwardingTransaction`. I dismissed it because I'm not sure if "standard account" is a good term either. It should probably be rather called "standard user account", and even that assumes that this contract will actually become "the" standard. But `StandardUserAccountTransaction` strikes me as too long.

General note on EIPs: As far as I know there are no EIPs for the sharding protocol itself. At least at the moment, I don't know if that will change.